src/entity/Entity.php

```diff
@@ -1080,6 +1081,13 @@ public function onCollideWithPlayer(Player $player) : void{
 
 	}
 
+	/**
+	 * Called when interacted or tapped by a Player
+	 */
+	public function onInteract(Player $player, Item $item, Vector3 $clickPos, int $slot) : bool{
```

Comment:
It's already capable of fetching this information (slot is always getItemInHandIndex()).

Reply:
perhaps I could just ditch passing item-related info altogether to `Entity::onInteract()` and `PlayerEntityInteractEvent`?